src/Model/Metadata.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\EuropaSearchClient\Model;
+
+/**
+ * A class that represents a document's metadata.
+ */
+class Metadata implements \JsonSerializable
```

Comment:
It would be more flexible we implement also `\ArrayAccess`. This would allow to use a Metadata instance as a simple array. Also we can extend `\Countable` so that we can make an elegant evaluation of the number of items.